tests/integration/bond_test.py

```diff
@@ -553,3 +556,57 @@ def test_set_miimon_100_on_existing_bond(bond99_with_2_slaves):
     bond_config[Bond.OPTIONS_SUBTREE] = {"miimon": 100}
     libnmstate.apply(state)
     assertlib.assert_state_match(state)
+
+
+@pytest.fixture
+def eth1_eth2_with_no_profile():
+    iface_down(ETH1)
+    iface_down(ETH2)
+    yield
+    iface_down(ETH1)
+    iface_down(ETH2)
+
+
+def _nmcli_profile_reactivate(ifname):
+    """
+    Use nmcli to deactivate the profile and reactivate
+    """
+    assert cmdlib.exec_cmd(["nmcli", "connection", "down", ifname])[0] == 0
+    # Wait slave been deactivated
+    time.sleep(1)
+    assert cmdlib.exec_cmd(["nmcli", "connection", "up", ifname])[0] == 0
+    # Wait slave been activated
+    time.sleep(1)
+
+
+def test_bond_mac_be_consistent_with_diff_slave_order(
```

Comment:
This tests only the order for two NICs when the bond is created, it would be great to also test the expected from creating a bond with one slave and adding a second and creating a bond with three NICs and then removing one, so we have clear definition of the expected behavior.

Reply:
When `eth2` is the first slave been added, the bond will use its mac address even after `eth1` been added afterwords. After reboot, bond will use eth1 MAC address.

To fix this in nmstate, nmstate should fully deactivate the bond which is unacceptable.

I think document this in nmstate github page is the only thing we could do

Reply:
@almusil what is your opinion on this?

@cathay4t can we add a warning when we hit the case that the MAC will change after reboot?

Reply:
If user don't include MAC in desire state that means they depending on kernel/NM/nmstate to decide. So I don't think we should warn them.

If mac address matters, they could define it in desire state.

Reply:
Ok, so the expectation is that the MAC address will not change for these these changes of the order of the slave NICs in the desired state:

Adding/removing:
- [eth2] -> [eth1, eth2]
- [eth1, eth2, eth3] -> [eth2, eth3]

Replacing:
- [eth1, eth2] -> [eth3]

Reordering:
- [eth1, eth2] -> [eth2, eth1]

But after a reboot, the MAC will change.

I am fine that we can also add the warning later, if you are against it. Somehow I think it makes sense to make this more explicit given that Nmstate is declarative. However, we can also table this discussion until we actually distinguish between runtime state and configured state. Nevertheless, it would be great to have tests for the described changes above IMHO.

Reply:
This change only handled the case where the slaves profile have not existed on the system before and fixed that issue. All other scenarios merit a test but not as part of this change.

I also expect some of them to fail, but I am unsure.